lib/oidcstrategy.js

```diff
@@ -504,10 +498,13 @@ Strategy.prototype.authenticate = function authenticateStrategy(req, options) {
    */ 
   const self = this;
 
+  var resource = options && options.resourceURL;
+  var customState = options && options.customState;
+
   // 'params': items we get from the request or metadata, such as id_token, code, policy, metadata, cacheKey, etc
   var params = {}; 
   // 'oauthConfig': items needed for oauth flow (like redirection, code redemption), such as token_endpoint, userinfo_endpoint, etc
-  var oauthConfig = {};
+  var oauthConfig = { 'resource': resource, 'customState': customState};
```

Comment:
So this make the `resource` and `state` configurable for each request, right? Meaning, for each incoming request, we can pass in different value, instead one-time only config during setup time.